crypto/evp/pmeth_fn.c

```diff
@@ -15,25 +15,19 @@
 #include "internal/evp_int.h"
 
 #define M_check_autoarg(ctx, arg, arglen, err) \
-        if (ctx->pmeth->flags & EVP_PKEY_FLAG_AUTOARGLEN) \
-                { \
-                size_t pksize = (size_t)EVP_PKEY_size(ctx->pkey); \
-                if (pksize == 0) \
-                        { \
-                        EVPerr(err, EVP_R_INVALID_KEY); /*ckerr_ignore*/\
-                        return 0; \
-                        } \
-                else if (!arg)                 \
-                        { \
-                        *arglen = pksize; \
-                        return 1; \
-                        } \
-                else if (*arglen < pksize) \
-                        { \
-                        EVPerr(err, EVP_R_BUFFER_TOO_SMALL); /*ckerr_ignore*/\
-                        return 0; \
-                        } \
-                }
+    if (ctx->pmeth->flags & EVP_PKEY_FLAG_AUTOARGLEN) {           \
+        size_t pksize = (size_t)EVP_PKEY_size(ctx->pkey);         \
+        if (pksize == 0) {                                        \
+            EVPerr(err, EVP_R_INVALID_KEY); /*ckerr_ignore*/      \
+            return 0;                                             \
+        } else if (!arg) {                                        \
```

Comment:
remove the else after return.  since this is all inside an if() statement, it's still one statement.
and in the else-if below.